### PR TITLE
IDGEN-42

### PR DESCRIPTION
**Pull request by mozzy11**
https://issues.openmrs.org/browse/IDGEN-42

### CODE REVIEW

**Comment by samuelmale** [comment]
Did you see the CI failure @mozzy11 ?

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierPoolResourceHandler.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+@SubClassHandler(supportedClass = IdentifierPool.class, supportedOpenmrsVersions = {"1.8.*", "1.9.*","1.10.*", "1.11.*","1.12.*"," 2.0.*"," 2.1.*", "2.2.*"})
+public class IdentifierPoolResourceHandler extends BaseDelegatingSubclassHandler<IdentifierSource, IdentifierPool>
+        implements DelegatingSubclassHandler<IdentifierSource, IdentifierPool> {
+
+
+
+    @Override
+    public String getResourceVersion() {
+        return "1.8";
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierPoolResourceHandler.java]: Isn't there a constant for this?

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.resource.api.SubResource;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@org.openmrs.module.webservices.rest.web.annotation.SubResource(parent = IdentifierSourceResource.class, path =
+        "identifier", supportedClass = String.class, supportedOpenmrsVersions = {"1.8.*", "1.9.*","1.10.*", "1.11.*","1.12.*"," 2.0.*"," 2.1.*", "2.2.*"})
+public class IdentifierResource implements SubResource {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java]: Why aren't you extending DelegatingSubResource?

File: omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    public static final String IDENTIFIER_KEY = "identifier";
+
+    @Override
+    public String getUri(Object instance) {
+        throw new UnsupportedOperationException();
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java]: Why?

File: api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java [hunk 4]
```diff
@@ -18,156 +18,39 @@
...
 import org.openmrs.PatientIdentifierType;
 import org.openmrs.User;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.openmrs.OpenmrsMetadata;
 
 /**
  * An IdentifierSource is a construct which can supply identifiers of a particular type
  */
-public abstract class IdentifierSource {
+public abstract class IdentifierSource implements OpenmrsMetadata {
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java]: Is this supposed be part of this ticket?

File: api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java [hunk 5]
```diff
@@ -18,156 +18,39 @@
...
     
     //****** PROPERTY ACCESS ******
     
-	/**
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java]: Did you format the code?

File: api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java (hunk 4 above)
Comment (wluyima) [reply api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java]: And this would need to extend BaseOpenmrsMetadata instead

File: api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java [hunk 6]
```diff
@@ -173,4 +173,20 @@ public String getBaseCharacterSet() {
 	public void setBaseCharacterSet(String baseCharacterSet) {
 		this.baseCharacterSet = baseCharacterSet;
 	}
+	
+	
+	
+	/**	
+	 * @return the nextSequenceValue	
+	 */	
+	public Long getNextSequenceValue() {
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java]: Why are these changes necessary?

**Comment by mozzy11** [comment]
alryt sir @wluyima , let me consider those comments as i try to update my PR

**Comment by mozzy11** [comment]
noo,  its just that dKayiwa wanted some some updates and i had to push some unfinished work ..just

**Comment by dkayiwa** [comment]
Closing this because the author unassigned himself from the ticket.
